src/errors.rs

```diff
@@ -313,5 +315,25 @@ error_chain! {
             description("Invalid Lines")
                 display("Invalid Lines")
         }
+
+        NoClickHouseClientAvailable {
+            description("The ClickHouse adapter has no client available")
+            display("The ClickHouse adapter has no client available")
+        }
+
+        ExpectedObjectEvent(found_type: ValueType) {
+            description("Expected object event")
+                display("Expected an object event, found a \"{found_type:?}\"")
+        }
+
+        MissingEventColumn(column: String) {
+            description("Missing event column")
+                display("Column \"{column}\" is missing")
+        }
+
+        UnexpectedEventFormat(column_name: String, expected_type: String, found_type: ValueType) {
```

Comment:
I feel like it's a bad smell to type `expected_type` as a `String`. The reason why I did that is because using `DummySqlType` would require it to be public, which looks a bit overkill to me.
Any opinion on this?

Reply:
This is fine. The fields inside those error types are only for conveying information to users, and you, as the error-type owner control how to use them. Also i think it is good practice, to keep the dependencies to other types low for those errors.

Reply:
Perfect. Thanks for your feedback!